app/lib/activitypub/activity/create.rb

```diff
@@ -5,10 +5,12 @@ class ActivityPub::Activity::Create < ActivityPub::Activity
   CONVERTED_TYPES = %w(Image Video Article Page).freeze
 
   def perform
-    return if delete_arrived_first?(object_uri) || unsupported_object_type? || invalid_origin?(@object['id'])
+    return if unsupported_object_type? || invalid_origin?(@object['id'])
 
     RedisLock.acquire(lock_options) do |lock|
       if lock.acquired?
+        return if delete_arrived_first?(object_uri)
```

Comment:
CodeClimate complains, but I find it clearer to just return here. Or maybe there's something I don't understand about Ruby blocks?

Reply:
I believe the code does what you want it to do, but in most cases return from a block is a mistake.